### PR TITLE
fix(cmake): make sure link flags are propagated to module library builds

### PR DESCRIPTION
**Pull request by jungleraptor**
`CMAKE_SHARED_LINKER_FLAGS_INIT` alone does not populate the link flags for module library targets, i.e.:

```cmake
add_library(foo MODULE)
```

This sets the variable in the crosstool script.

### CODE REVIEW

**Comment by jungleraptor** [comment]
All that's failing is the buildifier step but no logs indicating an issue with the code.

**Comment by jungleraptor** [comment]
Not sure why this got closed.

File: foreign_cc/private/cmake_script.bzl [hunk 1]
```diff
@@ -204,6 +204,7 @@ _CMAKE_CACHE_ENTRIES_CROSSTOOL = {
     "CMAKE_EXE_LINKER_FLAGS": struct(value = "CMAKE_EXE_LINKER_FLAGS_INIT", replace = False),
     "CMAKE_RANLIB": struct(value = "CMAKE_RANLIB", replace = True),
     "CMAKE_SHARED_LINKER_FLAGS": struct(value = "CMAKE_SHARED_LINKER_FLAGS_INIT", replace = False),
+    "CMAKE_MODULE_LINKER_FLAGS": struct(value = "CMAKE_MODULE_LINKER_FLAGS_INIT", replace = False),
```
Comment (jsharpe) [inline foreign_cc/private/cmake_script.bzl:207]:
```suggestion
    "CMAKE_MODULE_LINKER_FLAGS": struct(value = "CMAKE_MODULE_LINKER_FLAGS_INIT", replace = False),
    "CMAKE_RANLIB": struct(value = "CMAKE_RANLIB", replace = True),
    "CMAKE_SHARED_LINKER_FLAGS": struct(value = "CMAKE_SHARED_LINKER_FLAGS_INIT", replace = False),
```

File: test/cmake_text_tests.bzl [hunk 2]
```diff
@@ -104,6 +104,7 @@ def _fill_crossfile_from_toolchain_test(ctx):
         "CMAKE_C_FLAGS_INIT": "-cc-flag -gcc_toolchain cc-toolchain",
         "CMAKE_EXE_LINKER_FLAGS_INIT": "executable",
         "CMAKE_SHARED_LINKER_FLAGS_INIT": "shared1 shared2",
+        "CMAKE_MODULE_LINKER_FLAGS_INIT": "shared1 shared2",
         "CMAKE_SYSROOT": "/abc/sysroot",
```
Comment (jsharpe) [inline test/cmake_text_tests.bzl:108]:
```suggestion
        "CMAKE_EXE_LINKER_FLAGS_INIT": "executable",
        "CMAKE_MODULE_LINKER_FLAGS_INIT": "shared1 shared2",
        "CMAKE_SHARED_LINKER_FLAGS_INIT": "shared1 shared2",
        "CMAKE_SYSROOT": "/abc/sysroot",
```

File: test/cmake_text_tests.bzl [hunk 3]
```diff
@@ -173,6 +174,7 @@ def _reverse_descriptor_dict_test(ctx):
         "CMAKE_C_FLAGS_INIT": struct(value = "CMAKE_C_FLAGS", replace = False),
         "CMAKE_EXE_LINKER_FLAGS_INIT": struct(value = "CMAKE_EXE_LINKER_FLAGS", replace = False),
         "CMAKE_SHARED_LINKER_FLAGS_INIT": struct(value = "CMAKE_SHARED_LINKER_FLAGS", replace = False),
+        "CMAKE_MODULE_LINKER_FLAGS_INIT": struct(value = "CMAKE_MODULE_LINKER_FLAGS", replace = False),
         "CMAKE_STATIC_LINKER_FLAGS_INIT": struct(value = "CMAKE_STATIC_LINKER_FLAGS", replace = False),
```
Comment (jsharpe) [inline test/cmake_text_tests.bzl:178]:
```suggestion
        "CMAKE_EXE_LINKER_FLAGS_INIT": struct(value = "CMAKE_EXE_LINKER_FLAGS", replace = False),
        "CMAKE_MODULE_LINKER_FLAGS_INIT": struct(value = "CMAKE_MODULE_LINKER_FLAGS", replace = False),
        "CMAKE_SHARED_LINKER_FLAGS_INIT": struct(value = "CMAKE_SHARED_LINKER_FLAGS", replace = False),
        "CMAKE_STATIC_LINKER_FLAGS_INIT": struct(value = "CMAKE_STATIC_LINKER_FLAGS", replace = False),
```

**Review by jsharpe — APPROVED** [review]
This looks reasonable to me - the buildifier errors are lexocographical orderings - We just need to reorder some of the dictionary key entries.

**Comment by jungleraptor** [comment]
Thanks for your help getting this merged @jsharpe !

**Pull request merged**